ocean_lib/ocean/ocean_assets.py

```diff
@@ -322,48 +375,185 @@ def create(
-            flags = bytes([0])
-            document = ddo_bytes
+        _ = erc721_token.set_metadata(
+            metadata_state=0,
+            metadata_decryptor_url=provider_uri,
+            metadata_decryptor_address=publisher_wallet.address,
+            flags=flags,
+            data=document,
+            data_hash=ddo_hash,
+            data_proofs=[],
+            from_wallet=publisher_wallet,
+        )
 
-        # Only compression, not encrypted
-        elif compress_flag and not encrypt_flag:
-            flags = bytes([1])
-            # Compress DDO
-            document = lzma.compress(ddo_bytes)
+        # Fetch the asset on chain
+        asset = self._get_aquarius(self._metadata_cache_uri).wait_for_asset(did)
 
-        # Only encryption, not compressed
-        elif encrypt_flag and not compress_flag:
-            flags = bytes([2])
-            # Encrypt DDO
-            encrypt_response = DataServiceProvider.encrypt(
-                objects_to_encrypt=ddo_string,
-                encrypt_endpoint=f"{provider_uri}/api/services/encrypt",
+        return asset
+
+    def update(
+        self,
+        did: str,
+        new_metadata: dict,
+        publisher_wallet: Wallet,
+        delete_erc20_tokens: Optional[List[str]] = [],
+        encrypted_files: Optional[str] = None,
+        new_services: Optional[list] = [],
+        new_credentials: Optional[dict] = None,
+        provider_uri: Optional[str] = None,
+        new_erc20_tokens_data: Optional[List[ErcCreateData]] = [],
+        new_deployed_erc20_tokens: Optional[List[ERC20Token]] = [],
+        encrypt_flag: Optional[bool] = False,
+        compress_flag: Optional[bool] = False,
+    ) -> str:
+        """Update an asset on-chain.
+
+        :param did: The DID of the asset to update.
+        :param new_metadata: dict conforming to the Metadata accepted by Ocean Protocol.
+        :param publisher_wallet: Wallet of the publisher updating this asset.
+        :param delete_erc20_tokens: List of existing ERC20 tokens and related Services to delete.
+        :param encrypted_files: str of the files that need to be encrypted before updating.
+        :param new_services: list of new Service objects to add.
+        :param new_credentials: new_credentials necessary for the asset.
+        :param provider_uri: str URL of service provider. This will be used as base to construct the serviceEndpoint for the `access` (download) service
+        :param new_erc20_tokens_data: list of ERC20CreateData necessary for deploying ERC20 tokens for different services.
+        :param new_deployed_erc20_tokens: list of ERC20 tokens which are already deployed.
+        :param encrypt_flag: bool for encryption of the DDO.
+        :param compress_flag: bool for compression of the DDO.
+        :return: update tx_result
+        """
+
+        old_asset = self._get_aquarius(self._metadata_cache_uri).wait_for_asset(did)
+
+        assert old_asset, "Asset not found."
```

Comment:
actually this could be a possibility: what if the old metadata state is expired, and we are updating it to active? Aquarius won't serve this asset, but we still want to update it. Not sure if the case is valid, but it's a discussion worth having.